tarogarden/caretaker.go

```diff
@@ -323,10 +323,33 @@ func (b *BatchCaretaker) seedlingsToAssetSprouts(ctx context.Context,
 		}
 
 		var groupKey *asset.GroupKey
+		// If the seedling has a group key specified and emission
+		// enabled, that group key was validated earlier. We need to
+		// sign the new genesis with that group key.
+		if seedling.EnableEmission && seedling.hasGroupKey() {
+			groupKeys, err := b.cfg.Log.FetchAllGroups(ctx)
```

Comment:
Rather than fetching _all_ the groups, seems like we can either just:
  * Use the key they provided, things should fail it it doesn't exist 
  * Modify the query to fetch a specific group based on the _tweaked_ group key

Reply:
Will modify the query to add tweaked group key as an arg.